watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack_conc_recirc.py

```diff
@@ -0,0 +1,474 @@
+    iscale.set_scaling_factor(m.fs.EDstack.cell_length, 1)
+    iscale.set_scaling_factor(m.fs.EDstack.cell_pair_num, 0.1)
+    iscale.set_scaling_factor(m.fs.EDstack.voltage_applied, 1)
+    iscale.set_scaling_factor(m.fs.pump0.control_volume.work, 1e1)
+    iscale.set_scaling_factor(m.fs.pump1.control_volume.work, 1e1)
+    iscale.calculate_scaling_factors(m)
+
+    # populate intitial properties throughout the system
+    m.fs.feed.initialize(optarg=optarg)
+    propagate_state(m.fs.arc0)
+    m.fs.sepa0.initialize(optarg=optarg)
+    propagate_state(m.fs.arc1b)
+    m.fs.pump1.deltaP[0].fix(2e5)
+    m.fs.pump1.initialize()
+    m.fs.pump1.deltaP[0].unfix()
+    propagate_state(destination=m.fs.pump0.inlet, source=m.fs.pump1.inlet)
+    m.fs.pump0.deltaP[0].fix(2e5)
+    m.fs.pump0.initialize()
+    m.fs.pump0.deltaP[0].unfix()
+    propagate_state(m.fs.arc1f)
+    propagate_state(m.fs.arc3f)
+    m.fs.EDstack.initialize(optarg=optarg)
+    propagate_state(m.fs.arc4)
+    m.fs.prod.initialize(optarg=optarg)
+    propagate_state(m.fs.arc5)
+    m.fs.prod.initialize(optarg=optarg)
+    m.fs.sepa1.split_fraction[0, "to_conc_in1"].fix(
+        max(2 - value(m.fs.recovery_vol_H2O) ** -1, 1e-8)
+    )
+    m.fs.sepa1.initialize(optarg=optarg)
+    m.fs.sepa1.split_fraction[0, "to_conc_in1"].unfix()
+    propagate_state(m.fs.arc6)
+    m.fs.disp.initialize()
+    propagate_state(m.fs.arc7)
+    propagate_state(m.fs.arc3b, direction="backward")
+    propagate_state(m.fs.arc2)
+    m.fs.mix0.initialize(optarg=optarg)
+    m.fs.costing.initialize()
+
+    iscale.calculate_scaling_factors(m)
+
+
+def display_model_metrics(m):
+
+    print("---Report the ED unit---")
+
+    m.fs.EDstack.report()
+
+    print("---Flow properties in feed, product and disposal---")
+
+    fp = {
+        "Feed": [
+            value(m.fs.feed.properties[0].flow_vol_phase["Liq"]),
+            value(
+                sum(
+                    m.fs.feed.properties[0].conc_mass_phase_comp["Liq", j]
+                    for j in m.fs.properties.ion_set
+                )
+            ),
+        ],
+        "Product": [
+            value(m.fs.prod.properties[0].flow_vol_phase["Liq"]),
+            value(m.fs.product_salinity),
+        ],
+        "Disposal": [
+            value(m.fs.disp.properties[0].flow_vol_phase["Liq"]),
+            value(m.fs.disposal_salinity),
+        ],
+    }
+    fp_table = pd.DataFrame(
+        data=fp,
+        index=["Volume Flow Rate (m3/s)", "Total Ion Mass Concentration (kg/m3)"],
+    )
+    print(fp_table)
+
+    print("---Performance Metrics---")
+
+    pm_table = pd.DataFrame(
+        data=[
+            value(m.fs.recovery_vol_H2O),
+            value(m.fs.mem_area),
+            value(m.fs.EDstack.cell_pair_num),
+            value(m.fs.EDstack.cell_length),
+            value(m.fs.EDstack.voltage_applied[0]),
+            value(m.fs.costing.specific_energy_consumption),
+            value(m.fs.costing.LCOW),
+        ],
```

Comment:
For ED-related metrics, any reason why you don't use `report()` anywhere?

Reply:
The `report()` was added in the last revision.